ddtrace/profiling/event.py

```diff
@@ -45,28 +45,10 @@ class StackBasedEvent(SampleEvent):
     task_name = attr.ib(default=None)
     frames = attr.ib(default=None)
     nframes = attr.ib(default=None)
-    trace_id = attr.ib(default=None, type=typing.Optional[int])
-    span_id = attr.ib(default=None, type=typing.Optional[int])
-    trace_type = attr.ib(default=None, type=typing.Optional[str])
-    trace_resource = attr.ib(default=None, type=typing.Optional[str])
+    span = attr.ib(default=None, type=typing.Optional[ddspan.Span])
```

Comment:
what is the lifecycle of these events? since it will cause us to keep this span in memory longer, do we know how often we expect these spans to be held before being released to GC?

maybe this is a better question for @P403n1x87 ?

Reply:
My understanding is that they are collected until profiler sends profile to the backend - which is 1 minute by default

Reply:
Is there a limit to the number of events we collect in 1 minute? This could give us an idea of how many spans we could be holding onto in memory

Reply:
To the best of my understanding there is no limit on number of events - however many events can be collected for the same span.
So we are essentially looking at keeping ~all traces for 1 minute

Reply:
> So we are essentially looking at keeping ~all traces for 1 minute

Makes sense to me. This means there might be some considerable build-up of traces in memory, but I don't have a feel for what this actually means in terms of actual numbers for a typical scenario.